applications/RANSApplication/custom_elements/convection_diffusion_reaction_cross_wind_stabilized_element.h

```diff
@@ -0,0 +1,989 @@
+        if (rElementalDofList.size() != TNumNodes) {
+            rElementalDofList.resize(TNumNodes);
+        }
+
+        const Variable<double>& r_variable =
+            TConvectionDiffusionReactionData::GetScalarVariable();
+
+        for (unsigned int i = 0; i < TNumNodes; ++i) {
+            rElementalDofList[i] = Element::GetGeometry()[i].pGetDof(r_variable);
+        }
+    }
+
+    void GetValuesVector(
+        Vector& rValues,
+        int Step = 0) const override
+    {
+        this->GetFirstDerivativesVector(rValues, Step);
+    }
+
+    void GetFirstDerivativesVector(
+        Vector& rValues,
+        int Step = 0) const override
+    {
+        if (rValues.size() != TNumNodes) {
+            rValues.resize(TNumNodes, false);
+        }
+
+        const auto& r_geometry = this->GetGeometry();
+        const Variable<double>& r_variable =
+            TConvectionDiffusionReactionData::GetScalarVariable();
+
+        IndexType LocalIndex = 0;
+        for (IndexType i_node = 0; i_node < TNumNodes; ++i_node) {
+            rValues[LocalIndex++] =
+                r_geometry[i_node].FastGetSolutionStepValue(r_variable, Step);
+        }
+    }
+
+    void GetSecondDerivativesVector(
+        Vector& rValues,
+        int Step = 0) const override
+    {
+        if (rValues.size() != TNumNodes) {
+            rValues.resize(TNumNodes, false);
+        }
+
+        const auto& r_geometry = this->GetGeometry();
+        const Variable<double>& r_variable =
+            TConvectionDiffusionReactionData::GetScalarRateVariable();
+
+        IndexType LocalIndex = 0;
+        for (IndexType i_node = 0; i_node < TNumNodes; ++i_node) {
+            rValues[LocalIndex++] =
+                r_geometry[i_node].FastGetSolutionStepValue(r_variable, Step);
+        }
+    }
+
+    /**
+     * ELEMENTS inherited from this class have to implement next
+     * CalculateLocalSystem, CalculateLeftHandSide and CalculateRightHandSide
+     * methods they can be managed internally with a private method to do the
+     * same calculations only once: MANDATORY
+     */
+
+    /**
+     * this is called during the assembling process in order
+     * to calculate all elemental contributions to the global system
+     * matrix and the right hand side
+     * @param rLeftHandSideMatrix: the elemental left hand side matrix
+     * @param rRightHandSideVector: the elemental right hand side
+     * @param rCurrentProcessInfo: the current process info instance
+     */
+    void CalculateLocalSystem(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        const ProcessInfo& rCurrentProcessInfo) override
+    {
+        // Check sizes and initialize matrix
+        if (rLeftHandSideMatrix.size1() != TNumNodes ||
+            rLeftHandSideMatrix.size2() != TNumNodes) {
+            rLeftHandSideMatrix.resize(TNumNodes, TNumNodes, false);
+        }
+
+        noalias(rLeftHandSideMatrix) = ZeroMatrix(TNumNodes, TNumNodes);
```

Comment:
Why is the LHS always zero?

Reply:
LHS is calcualted from CalculateLocalVelocityContribution method since anyways RHS calculation requires LHS (so to avoid calculating LHS twice). I think this is the same methodology followed in fluid ryt? :|

Reply:
In the "old" elements we do it in that way. In the future we can retake the issue, as I don't think it is theoretically needed.

Reply:
What do you propose?